### PR TITLE
Eris Mech Port Project

### PR DESCRIPTION
**Pull request by Stutternov**
## About The Pull Request
Tl;dr - same project as https://github.com/sojourn-13/sojourn-station/pull/3585 - attempting it again since mechs are still not fully fixed and looking more at the system is it incredibly out of date and antiquated compared to what exists on Bay/Eris. Read that closed PRs reasoning and outlines for why I'm doing the port again. **This is a WIP so it will stay drafted till it's finished. Contribution and help with solving bugs/porting our unique mech code atop of it will be helpful.**

## Changelog
:cl:
add: Too much to put; modularized mech code ported from Bay to Eris, and now an attempt of porting it to here.
add: Reworked research line for mechs, new equipment and a few new weapons or modules. New mech parts and designs.
/:cl:

### CODE REVIEW

**Comment by Trilbyspaceclone** [comment]
This ruins something someware I think idk didnt read the code or pr
its p-bad i dislike it
Please read contributing guidelines please

**Comment by Stutternov** [comment]
> This ruins something someware I think idk didnt read the code or pr its p-bad i dislike it Please read contributing guidelines please

https://user-images.githubusercontent.com/47883419/176964115-14261671-3a13-4d1e-b1c2-20a31e0432c4.mp4

**Comment by JohnWilsonDeez** [comment]
This is bad and I hate it oh wait just read it :)

**Comment by Lando-o-Rando** [comment]
Kill yourself, now put in my sprite >:)

**Comment by Stutternov** [comment]
The work required to fully port this is kind of beyond me at this point. We have so much snowflake mech stuff, that we claim we need to keep - yet also bitch how useless it is - that it is a pain in the ass to even attempt to port everything wholesale.

The mech system is much better than what we have but since I do not want to play games with developers/staff with 'WE HAVE TO KEEP EVERYTHING WE HAVE!' - I'm ceasing port work for this.
